src/libcollections/str.rs

```diff
@@ -10,9 +10,28 @@
 
 //! Unicode string slices.
 //!
+//! The `&str` type is one of the two main string types, the other being `String`.
+//! Unlike its `String` counterpart, its contents are borrowed and therefore
+//! cannot be moved someplace else.
+//!
+//! # Basic Usage
+//! A basic string declaration of `&str` type:
```

Comment:
could you insert a `//!` above this line, to give it some space?

Also, I might say "Defining a variable with the type `&str`" here, sounds a teeny bit more natural.